### ISSUE TITLE
a couple of notes on loss and activation

### ISSUE THREAD

**Issue by mahmoudyusof** [AryanSethi/Deblurring_autoencoder#1]
Hello, again. I've noticed your progress through the past few days, and I thought I could help. I've noticed your choice of loss function and last layer activation are `binary_crossentropy` and `sigmoid` respectively. however this combination is used for binary classification tasks, which is not what you're doing. your output is not a probability distribution nor are you solving a classification problem. a more appropriate loss function would be mean squared error or `mse` for short since your outputs are linear values 
(0 - 255 pixel data), now that I've mentioned linearity, it's time to say that the last layer's activation should be linear (aka no activation at the end of the network). Now to evaluate your model, you need to look at the loss, not the accuracy. accuracy is only meaningful in classification problems. I would encourage you to read some other open source projects that do the same task and learn from them, it will be very beneficial. best of luck to you.

**Comment by AryanSethi** [comment]
Hello Mahmoud, I'm so glad you were finding time to check on my project.
I was not aware of most of the facts that you mentioned here. This opens up
alot to research about. I didn't even know that it was more of a 'need'
than a 'choice' to use the right loss function.
This explains the weird weight and accuracy values I was getting during
training.
I'll update my code as you suggested.
*Thanks a lot.*

**Comment by AryanSethi** [comment]
I performed similar operations with smaller target size and no activation at the end. Getting significantly low loss now.
Thanks @mahmoudyusof
